core/Archive/DataTableFactory.php

```diff
@@ -157,21 +161,65 @@ public function useSubtable($idSubtable)
      */
     public function make($index, $resultIndices)
     {
+        $keyMetadata = $this->getDefaultMetadata();
```

Comment:
A lot of time was spent in setting and transforming metadata. This is much faster now. Eg before we did set a table metadata like `array('period' => '2012-12...', 'site' => 5)` and they were afterwards "transformed" to `array('period' => Period, 'site' => new Site())`.